File: Python/ShortestandLexicographicallySmallestBeautifulString.py
class Solution:
    def shortestBeautifulSubstring(self, s: str, k: int) -> str:
        result = ""

        for i in range(0, len(s)):
            ones = 0

            for j in range(i, len(s)):
                if s[j] == "1": ones += 1

                if ones == k:
                    substring = s[i:j+1]

                    if result == "" or len(substring) < len(result) or (len(substring) == len(result) and substring < result):
                        result = substring

                    break

        return result

solution = Solution()

print(solution.shortestBeautifulSubstring("100011001", 3))

File: JavaScript/ShortestandLexicographicallySmallestBeautifulString.js
const shortestBeautifulSubstring = function (s, k) {
  let result = "";

  for (let i = 0; i < s.length; i++) {
    let ones = 0;

    for (let j = i; j < s.length; j++) {
      if (s[j] === "1") ones++;

      if (ones === k) {
        const substring = s.slice(i, j + 1);

        if (
          result === "" ||
          substring.length < result.length ||
          (substring.length === result.length && substring < result)
        ) {
          result = substring;
        }
        break;
      }
    }
  }

  return result;
};
console.log(shortestBeautifulSubstring("100011001", 3));
